### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by kkimdev** [eslint/eslint#17499]
### What rule do you want to change?

require-await

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
abstract class A {
  abstract func(): Promise<undefined>;
}

class B extends A {
  override async func() {
    return undefined;
  }
}
```

### What does the rule currently do for this code?

Async method 'func' has no 'await' expression.eslint[@typescript-eslint/require-await](https://typescript-eslint.io/rules/require-await)

### What will the rule do after it's changed?

`@typescript-eslint/require-await` will have an option "AllowInOverriddenFunctions".  When it's `true`, it won't warn `async` function without `await` inside if `override` is specified.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
> `@typescript-eslint/require-await` will have an option "AllowInOverriddenFunctions".

For proposals related to typescript-eslint plugin rules, please submit issues in the typescript-eslint repo:

https://github.com/typescript-eslint/typescript-eslint
